tensorboard/plugins/histogram/metadata.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Information about histogram summaries."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import json
+
+import tensorflow as tf
+
+
+PLUGIN_NAME = 'histograms'
+
+
+# TODO(@wchargin,@dandelionmane): Once `display_name` and
+# `summary_description` land in core, use those first-class entities
+# instead.
+HistogramMetadata = collections.namedtuple(
+  'HistogramMetadata', ('display_name', 'description'))
+
+
+def create_summary_metadata(display_name, description):
+  content = HistogramMetadata(display_name=display_name or '',
+                              description=description or '')
+  metadata = tf.SummaryMetadata()
+  metadata.plugin_data.add(
+    plugin_name=PLUGIN_NAME,
+    content=json.dumps(content._asdict()),  # pylint: disable=protected-access
```

Comment:
(a) It looks from docs and context like `json_format` is just for protocol buffers, but `content` is a `dict` (`namedtuple`). Would that even work? (b) Why? Is the result different? Why introduce a new abstraction in addition to the standard library's `json` module?